onadata/apps/logger/models/attachment.py

```diff
@@ -86,3 +90,12 @@ def file_hash(self):
     def filename(self):
         if self.media_file:
             return os.path.basename(self.media_file.name)
+
+    @classmethod
+    def soft_delete(cls, queryset, user=None):
+        """
+        Soft deletes an attachment by adding a deleted_at timestamp.
+        """
+        queryset.update(deleted_at=timezone.now())
+        if user:
+            queryset.update(deleted_by=user)
```

Comment:
Can the above function be edited to take an instance instead of a queryset? Then we can use the `cls` object?
```python
def soft_delete(cls, instance, user=None):
    ...
    queryset = cls.objects.filter(name__in=instance.get_expected_media())
    kwargs = {'deleted_at': timezone.now()}
    if user:
        kwargs.update({'deleted_by': user})
    queryset.update(**kwargs)
```

This would then the classmethod would be called as:
```python
...
Attachment.soft_delete(instance)
```

Reply:
well actually, I didn't realize this was relative to an instance when I suggested adding a class method, so that's my mistake.

If this is always relative to an instance, I'd think it's better as an instance method, and we'd call with `instance.soft_delete()`, have `queryset = self.__class__.objects.filter(name__in=self.get_expexted_media())`

Is there a problem w/that?